astropy/time/formats.py

```diff
@@ -731,17 +731,28 @@ def parse_string(self, timestr, subfmts):
             raise ValueError('Time {0} does not match {1} format'
                              .format(timestr, self.name))
 
+    @staticmethod
+    def _decode_if_needed(val, encoding='ascii'):
+        """Decode ``vall`` if it is a bytes instance."""
+        if type(result) is bytes:
+            result = result.decode(encoding)
+        return result
+
     def set_jds(self, val1, val2):
         """Parse the time strings contained in val1 and set jd1, jd2"""
         # Select subformats based on current self.in_subfmt
         subfmts = self._select_subfmts(self.in_subfmt)
 
+        # We cannot just use the iterator over val1, since that effectively
+        # treats it as a regular array, thus bypassing the bytes to string
+        # conversion in Column. Hence, we create the index and use that
+        # inside the loop.
         iterator = np.nditer([val1, None, None, None, None, None, None],
                              op_dtypes=[val1.dtype] + 5*[np.intc] + [np.double])
 
         for val, iy, im, id, ihr, imin, dsec in iterator:
             iy[...], im[...], id[...], ihr[...], imin[...], dsec[...] = (
-                self.parse_string(val.item(), subfmts))
+                self.parse_string(self._decode_if_needed(val.item()), subfmts))
```

Comment:
Is there a noticeable performance hit to adding this extra call within the inner loop?

Reply:
One point is that you know in advance from `val1.dtype` whether *every* value needs decoding or *none* need decoding, so there is some opportunity for optimization.  But maybe that is truly in the noise and code clarity is more important.